### ISSUE TITLE
Proxy authentication don't use rolesHeader

### ISSUE THREAD

**Issue by kiblik** [floragunncom/search-guard#162]
I setup search-guard to use XFF and proxy authentication:

```
searchguard:
  dynamic:
    http:
      anonymous_auth_enabled: false
      xff:
        enabled: true
        internalProxies: '192\.168\.1\.*|127\.*' # regex pattern
        remoteIpHeader:  'x-forwarded-for'
        proxiesHeader:   'x-forwarded-by'
    authc:
      proxy_auth_domain:
        enabled: true
        order: 1
        http_authenticator:
          type: proxy
          challenge: false
          config:
            user_header: "x-proxy-user"
            roles_header: "x-proxy-roles"
        authentication_backend:
          type: noop

```
I tested this setup using curl:

`curl -XGET 'http://192.168.1.21:9200/_searchguard/authinfo?pretty=true' -v -H "x-proxy-user: bfu" -H "x-proxy-roles: testRole" -H "x-forwarded-for: 192.168.1.21"`
```
* Hostname was NOT found in DNS cache
*   Trying 192.168.1.21...
* Connected to 192.168.1.21 (192.168.1.21) port 9200 (#0)
> GET /_searchguard/authinfo?pretty=true HTTP/1.1
> User-Agent: curl/7.35.0
> Host: 192.168.1.21:9200
> Accept: */*
> x-proxy-user: bfu
> x-proxy-roles: testRole
> x-forwarded-for: 192.168.1.21
>
< HTTP/1.1 200 OK
< Content-Type: application/json; charset=UTF-8
< Content-Length: 169
<
{
  "user" : "User [name=bfu, roles=[]]",
  "remote_address" : "192.168.1.21:41942",
  "sg_roles" : [ "sg_public" ],
  "principal" : null,
  "peer_certificates" : "0"
}
* Connection #0 to host 192.168.1.21 left intact
```

ES user is authenticated but list of roles is empty:

```
[2016-07-06 14:11:09,922][DEBUG][com.floragunn.searchguard.http.XFFResolver] xff resolve /192.168.1.21:41991
[2016-07-06 14:11:09,922][DEBUG][com.floragunn.searchguard.http.XFFResolver] xff resolved /192.168.1.21:41991 to /192.168.1.21:41991
[2016-07-06 14:11:09,923][DEBUG][com.floragunn.searchguard.auth.BackendRegistry] Try to extract auth creds from http proxy
[2016-07-06 14:11:09,923][DEBUG][com.floragunn.searchguard.http.HTTPProxyAuthenticator] headers [User-Agent=curl/7.35.0, Host=192.168.1.21:9200, Accept=*/*, x-proxy-user=bfu, x-proxy-roles=testRole, x-forwarded-for=192.168.1.21]
[2016-07-06 14:11:09,923][DEBUG][com.floragunn.searchguard.http.HTTPProxyAuthenticator] userHeader x-proxy-user, value bfu
[2016-07-06 14:11:09,923][DEBUG][com.floragunn.searchguard.http.HTTPProxyAuthenticator] rolesHeader x-proxy-roles, value testRole
[2016-07-06 14:11:09,923][DEBUG][com.floragunn.searchguard.auth.BackendRegistry] User 'bfu' is in cache? true (cache size: 2)
[2016-07-06 14:11:09,923][DEBUG][com.floragunn.searchguard.auth.BackendRegistry] User 'User [name=bfu, roles=[]]' is authenticated

```
ES version: 2.3.3
SG version: 2.3.3
OpenJDK Runtime Environment (IcedTea 2.6.6) (7u101-2.6.6-0ubuntu0.14.04.1)
OpenJDK 64-Bit Server VM (build 24.95-b01, mixed mode)
Ubuntu 14.04.4 LTS
Cluster:
- 1x master
- 2x master+data

**Comment by floragunncom** [comment]
maybe you hit #150 which is fixed in Search Guard 2.3.3.2

**Comment by kiblik** [comment]
Yes, it works. Thankx
